sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -78,6 +79,20 @@ case class SubExprEliminationState(isNull: String, value: String)
 case class SubExprCodes(codes: Seq[String], states: Map[Expression, SubExprEliminationState])
 
 /**
+ * The main information about a new added function.
+ *
+ * @param functionName String representing the name of the function
+ * @param subclassName Optional value which is empty if the function is added to
+ *                     the outer class, otherwise it contains the name of the
+ *                     inner class in which the function has been added.
+ * @param subclassInstance Optional value which is empty if the function is added to
+ *                         the outer class, otherwise it contains the name of the
+ *                         instance of the inner class in the outer class.
+ */
+private[codegen] case class NewFunction(functionName: String, subclassName: Option[String],
+    subclassInstance: Option[String])
```

Comment:
The indent issues.

Reply:
may I ask you to kindly explain me which is the right indentation in this case? Thanks.

Reply:
I think that [2-space indentation](https://github.com/databricks/scala-style-guide#indent) is expected.

Reply:
Thanks for the link @kiszk , but it says that for parameters a 4 space indent is required: look at the example with `class Foo`.

Reply:
IIUC, 4-space indentation is used when it is followed by 2-space indentation. Here is [an example](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala#L49).

Reply:
In this example, there is 2 space indentation because there is a `extends` and according to the link you provided me, the `extends` requires a 2 space indentation, while the parameters a 4 space one.

Reply:
I see. This is just my understanding.  
Let us wait for explanation fro @gatorsmile.

Reply:
See the link: https://github.com/databricks/scala-style-guide#spacing-and-indentation

If it does not fit one line, we do it like
```Scala
private[codegen] case class NewFunction(
    functionName: String,
    subclassName: Option[String],
    subclassInstance: Option[String])
```